### PR TITLE
Refactor manifests processing.

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR:
1. Removed all manifests that are not used now, we will add them incrementally.
2. Added a yaml config where we can list supported providers and read it for installing them.
3. Added `operator-components.yaml` as temporary storage before we get upstream repo setup.
4. Most important, refactor manifests processing. With this change, we can have a consistent way for all providers to import their manifests to openshift. The import-assets binary reads assets from github, replaces cert-manager with service-ca, places CRDs and RBAC in manifests directory, all other manifests will be placed in assets to be managed by upstream CAPI operator.

### CODE REVIEW

**Review by elmiko — COMMENTED** [review]
i think this generally looks good @alexander-demichev , i have one question inline and i am also thinking we should add a README to the `hack/import-assets` directory. the readme could go into more detail about how the tool works. i see the note in the main readme, but it makes me wonder if we shouldn't have deeper instructions for people who have to add providers in the future.

File: hack/import-assets/operator-components.yaml [hunk 1]
```diff
@@ -0,0 +1,6115 @@
...
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .spec.version
+      name: Version
+      type: string
+    - jsonPath: .status.conditions[?(@.type=='Ready')].status
+      name: Ready
+      type: string
+    name: v1alpha1
```
[the remark this answers is not in the document]
Comment (alexander-demichev) [reply hack/import-assets/operator-components.yaml]: Yes, we are creating v1alpha1 objects. In upstream we decided to start iterating operator API from v1aplha1, I believe that by the time when CAPI in openshift reaches GA, operator API should be much more mature. I think now we are the most active developers of it, we should make sure that it stays compatible with future versions.

**Comment by elmiko** [comment]
thanks for the answer @alexander-demichev , one other question, does the readme need updating (or add a readme to the import-assets dir) for the new changes?

File: hack/import-assets/customizations.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+	}
+	return finalObjs
+}
+
+func findWebhookServiceSecretName(objs []unstructured.Unstructured) map[string]string {
+	serviceSecretNames := map[string]string{}
+	certSecretNames := map[string]string{}
+
+	secretFromCertNN := func(certNN string) (string, bool) {
+		certName := strings.Split(certNN, "/")[1]
```
[the remark this answers is not in the document]
Comment (alexander-demichev) [reply hack/import-assets/customizations.go:84]: yes, I didn't look much into that code. It was copied from another file. Let me fix it.

File: hack/import-assets/customizations.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+	finalObjs := []unstructured.Unstructured{}
+	rbacObjs := []unstructured.Unstructured{}
+	crdObjs := []unstructured.Unstructured{}
+	for _, obj := range objs {
+		switch obj.GetKind() {
+		case "ClusterRole", "Role", "ClusterRoleBinding", "RoleBinding", "ServiceAccount":
+			setOpenShiftAnnotations(obj, false)
+			setTechPreviewAnnotation(obj)
+			rbacObjs = append(rbacObjs, obj)
+		case "CustomResourceDefinition":
```
[the remark this answers is not in the document]
Comment (alexander-demichev) [reply hack/import-assets/customizations.go:158]: I think they do

File: hack/import-assets/customizations.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+	return append(bytes.TrimRight(b, "\n"), []byte("\n")...)
+}
+
+func writeRBACComponentsToManifests(fileName string, objs []unstructured.Unstructured) error {
+	return writeComponentsToManifests(fileName, objs)
+}
+
+func writeCRDComponentsToManifests(fileName string, objs []unstructured.Unstructured) error {
+	return writeComponentsToManifests(fileName, objs)
+}
```
[the remark this answers is not in the document]
Comment (alexander-demichev) [reply hack/import-assets/customizations.go:179]: Leftover after refactoring :D I was too tired that day

File: providers-list.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+#   version: "v0.5.2"
+# - name: "metal3"
+#   type: "InfrastructureProvider"
+#   version: "v0.5.2"
+# - name: "gcp"
+#   type: "InfrastructureProvider"
+#   version: "v0.4.0"
+# - name: "openstack"
+#   type: "InfrastructureProvider"
+#   version: "v0.4.0"
```
[the remark this answers is not in the document]
Comment (alexander-demichev) [reply providers-list.yaml:18]: yes

**Comment by alexander-demichev** [comment]
@elmiko I plan to add a readme once we test this on a provider we own, to be sure that this approach actually work and everyone is happy with it.

**Comment by elmiko** [comment]
> @elmiko I plan to add a readme once we test this on a provider we own, to be sure that this approach actually works and everyone is happy with it.

that sounds great, thanks @alexander-demichev 
/lgtm

**Comment by JoelSpeed** [comment]
/hold There are legitimate issues within the tests that will need further investigation

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-capi-operator/23/pull-ci-openshift-cluster-capi-operator-main-e2e-aws-capi-techpreview/1471745630392029184

**Pull request merged**